src/Core/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -284,7 +287,9 @@ private OpenApiPaths BuildPaths()
                 // The OpenApiResponses dictionary key represents the integer value of the HttpStatusCode,
                 // which is returned when using Enum.ToString("D").
                 // The "D" format specified "displays the enumeration entry as an integer value in the shortest representation possible."
+                // It will only contain $select query parameter to allow the user to specify which fields to return.
                 OpenApiOperation getOperation = CreateBaseOperation(description: GETONE_DESCRIPTION, tags: tags);
+                getOperation.Parameters = new List<OpenApiParameter>() { _tableAndViewSelectQueryParameter };
```

Comment:
As discussed here: https://github.com/Azure/data-api-builder/pull/1769#discussion_r1380877727

we only provide `$select` for get by id as When we're retrieving a single resource by its ID, other parameters like `$filter`,`$items`, `$after` generally don't make sense as there's only one resource to retrieve, so there's no need to filter or sort the results, and it's usually desirable to return all of the resource's fields.

However, there might be some cases where $select could be useful in a getById request if the resource has a large number of fields and we only need a few of them

Reply:
Even if they are not applicable, do we allow them by definition or error out in validation? If we allow them, I think the OpenAPI doc should mention those. 
To me, $filter is also applicable, I want to getbyId but filter out if that item doesnt satisfy some condition.

Reply:
The purpose of a getByID call is to retrieve a single, specific resource based on its unique identifier. what would we filter on a single resource? this would just not result in any output if the condition fails

also in case we would want to do it, we could achive this in getALL call as well by addding a filter on id with the condition. So, this wouldn't add any value

Reply:
GetbyId only if that id belongs to a certain country for e.g.

If we are thinking of usefulness, GetALLfilterOnId = GetById anyway, so how is GetById useful? 
We cant determine what might be useful to developers, in some cases GetById with filter could be useful in others, GetAllFilteronId+Othercondition. 

Irrespective of usefulness, the OpenAPI doc is to provide information about what APIs are available. If we support $filter on getById today, we should mention it in OpenAPI. But if the engine itself blocks support for it already, then we dont need to implement it now.